q2_types_genomics/reference_db/_format.py

```diff
@@ -5,17 +5,81 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
-
-
+import re
 from qiime2.plugin import model
+from qiime2.core.exceptions import ValidationError
 from q2_types_genomics.plugin_setup import plugin
-from q2_types_genomics.reference_db._type import ReferenceDB, Eggnog, Diamond
+from q2_types_genomics.reference_db._type import (
+    ReferenceDB, Eggnog, Diamond, EggnogSequenceTaxa
+)
+from q2_types.feature_data import MixedCaseProteinFASTAFormat
 
 
 class EggnogRefTextFileFmt(model.TextFileFormat):
+    _expected_columns = [
+            '# Taxid',
+            'Sci.Name',
+            'Rank',
+            'Named Lineage',
+            'Taxid Lineage'
+    ]
+    _line_pattern = re.compile(
+        r'^\d+\t'  # Taxid
+        r'([^\t]*\t)'  # Sci.Name
+        r'(no rank|species|subspecies)\t'  # Rank
+        r'([^\t]*\t)'  # Named Lineage
+        r'\d+(,\d+)*$'  # 'Taxid Lineage'
+    )
+
+    def _validate_1st_line(self, line):
+        fields = line.strip("\n").split("\t")
+        if len(fields) > 5:
+            raise ValidationError(
+                "Too many columns.\n"
+                "Expected columns:\n"
+                f"{self._expected_columns}\n"
+                "Columns given:\n"
+                f"{fields}"
+            )
+
+        if not (
+            fields[0] == '# Taxid' and
+            fields[1] == 'Sci.Name' and
+            fields[2] == 'Rank' and
+            fields[3] == 'Named Lineage' and
+            fields[4] == 'Taxid Lineage'
+        ):
+            raise ValidationError(
+                "Wrong columns.\n"
+                "Expected columns:\n"
+                f"{self._expected_columns}\n"
+                "Columns given:\n"
+                f"{fields}"
```

Comment:
When you insert lists like that, maybe you could join them with a `,` for the message to look a bit prettier?

Reply:
I think it looks pretty neat in the message:

```bash
Wrong columns.
Expected columns:
['# Taxid', 'Sci.Name', 'Rank', 'Named Lineage', 'Taxid Lineage']
Columns given:
['# Taxid', 'Sci.Name', 'Rank', 'Named Lineage', 'Taxid Lineagee']
```